Added Grammars

hilliardj opened the pull request:
Added the Grammar from the zip file submitted, and added a proposed grammar that removes left recursion.

hilliardj committed:
Added Grammars

Added the Grammar from the zip file submitted, and added a proposed grammar that removes left recursion.

hilliardj commented:
@justiceadamsUNI @harshavarmak Let me know what you think of the updated grammar removing left recursion.  Thanks!

justiceadamsUNI commented on Compiler/doc/languageGrammar.txt:
```diff
@@ -0,0 +1,60 @@
+            <ACTUALS> ::= e
+                        | <NONEMPTYACTUALS>
+
+    <NONEMPTYACTUALS> ::= <EXPR>
```
This NONEMPTYACTUALS has  <EXP> on both sides of the | so it's ambiguous still. There's quite a few places this happens yet (SIMPLE-EXPR, TERM, NONEMPTYFORMALS)  So we need to fix that.

justiceadamsUNI commented:
Any reason there's 3 different revised Grammars. I vote we just have orignalGrammar.txt and RevisedGrammar.txt. Then they can be edited anywhere with any editor. We can work on this tomorrow

hilliardj committed:
Removed PDF & DOCX File

Also updated Grammar file to finish up with the left recursion/left factoring issues.

justice-adams commented:
We're gonna wait for the OK from wallingford after the status check to merge this

justiceadamsUNI commented:
Waiting for wallingford to give the OK before we merge this

hilliardj committed:
Explaination of Grammar

Explaination for theRevised Grammar

hilliardj committed:
Made Minor Tweeks

Made minor changes to the Grammar Revisions Explaination

justiceadamsUNI commented:
I read through this and I'm on board. Nice job @hilliardj . We can merge this once wallingford says it's good.

hilliardj committed:
First Sets, Follow Sets, and Parse Table Excel File

Added the First Sets image, Follow Sets image, and Excel file graphic of the parse table

harshavarmak committed:
Added the pretty print of the First and Follow set

harshavarmak commented:
@hilliardj I've changed two of the First set, the First(Program) and First(Definitions), it should be only {$}

harshavarmak committed:
Changed a couple of First to make it right

harshavarmak committed:
Lots of corrections to showcase the real First and Follow set
